src/operations/common_functions.ts

```diff
@@ -81,19 +82,23 @@ export function indexInformation(
     });
 }
 
-export function prepareDocs(coll: any, docs: any, options: any) {
+export function prepareDocs(
+  coll: Collection,
+  docs: Document[],
+  options: { forceServerObjectId?: boolean }
+): Document[] {
   const forceServerObjectId =
     typeof options.forceServerObjectId === 'boolean'
       ? options.forceServerObjectId
-      : coll.s.db.options.forceServerObjectId;
+      : coll.s.db.options?.forceServerObjectId;
 
   // no need to modify the docs if server sets the ObjectId
   if (forceServerObjectId === true) {
     return docs;
   }
 
-  return docs.map((doc: any) => {
-    if (forceServerObjectId !== true && doc._id == null) {
+  return docs.map(doc => {
+    if (doc._id == null) {
```

Comment:
`forceServerObjectId` is either false or undefined after the check above asserts it can't be `true` so this was redundant